pkg/commands/deploy.go

```diff
@@ -253,6 +295,12 @@ func deployOnCluster(commonOpts *CommonOptions, opts *DeployOptions) error {
 	if opts.clusterPlatform == platform.Unknown {
 		return fmt.Errorf("cannot autodetect the platform, and no platform given")
 	}
+	versionDetect, source := detect.FindVersion(platDetect.Discovered, commonOpts.UserPlatformVersion)
+	commonOpts.DebugLog.Printf("Version detection source: %s", source)
+	opts.clusterVersion = versionDetect.Discovered
+	if opts.clusterVersion == platform.MissingVersion {
+		return fmt.Errorf("cannot autodetect the platform version, and no version given")
+	}
```

Comment:
please pass `Version: opts.ClusterVersion` down when we call `updaters.Deploy` (you may need to change `updaters.Options`)